src/preflight.js

```diff
@@ -3,42 +3,70 @@ var constants = require('./constants.js')
 
 exports.handler = function (options) {
   var matcher = originMatcher.create(options.origins)
-  return function (req, res, next) {
-    if (req.method !== 'OPTIONS') return next()
 
-    // 6.2.1 and 6.2.2
-    var originHeader = req.headers['origin']
-    if (!matcher(originHeader)) return next()
+  if (originMatcher.generic(options.origins)) {
```

Comment:
Same comment as above, but to a much greater extent. Is there a way we can do this that doesn't treat the whole block as a special case? We are adding an incredibly amount of complexity here to support this.